sound/soc/sof/ipc.c

```diff
@@ -350,53 +291,19 @@ static void sof_ipc_tx_msg_reply_complete(struct snd_sof_ipc *ipc,
 	wake_up(&msg->waitq);
 }
 
-/* drop all IPC messages in preparation for DSP stall/reset */
-void sof_ipc_drop_all(struct snd_sof_ipc *ipc)
-{
-	struct snd_sof_dev *sdev = ipc->sdev;
-	struct snd_sof_ipc_msg *msg, *tmp;
-
-	/* drop all TX and Rx messages before we stall + reset DSP */
-	spin_lock_irq(&sdev->ipc_lock);
-
-	list_for_each_entry_safe(msg, tmp, &ipc->tx_list, list) {
-		list_move(&msg->list, &ipc->empty_list);
-		dev_err(sdev->dev, "error: dropped msg %d\n", msg->header);
-	}
-
-	list_for_each_entry_safe(msg, tmp, &ipc->reply_list, list) {
-		list_move(&msg->list, &ipc->empty_list);
-		dev_err(sdev->dev, "error: dropped reply %d\n", msg->header);
-	}
-
-	spin_unlock_irq(&sdev->ipc_lock);
-}
-EXPORT_SYMBOL(sof_ipc_drop_all);
-
 /* handle reply message from DSP */
 int snd_sof_ipc_reply(struct snd_sof_dev *sdev, u32 msg_id)
 {
-	struct snd_sof_ipc_msg *msg;
-
-	msg = sof_ipc_reply_find_msg(sdev->ipc, msg_id);
-	if (!msg) {
-		dev_err(sdev->dev, "error: can't find message header 0x%x",
-			msg_id);
-		return -EINVAL;
-	}
-
 	/* wake up and return the error if we have waiters on this message ? */
-	sof_ipc_tx_msg_reply_complete(sdev->ipc, msg);
+	sof_ipc_tx_msg_reply_complete(sdev->ipc, &sdev->ipc->msg);
+
 	return 0;
```

Comment:
Now snd_sof_ipc_reply alway reply 0.  Actually there is another effect for original code return -EINVAL.
It is for ROM message which is not in our msg list.  There is a workaround on up2( I mentioned msg != 0x1004000) , so you don't get any issue, but the workaround is not on other platforms.

Reply:
As mentioned above, I think that's wrong. snd_sof_ipc_reply() has to distinguish between replies and new messages from the DSP. I'll restore that function.

Reply:
I don't understand this case. If this is for a ROM message, that isn't on the list, so, that isn't a response, that's an asynchronous message from the ROM, right? Then why does the HDA IPC IRQ handler recognises it as a reply: "is this a reply message from the DSP?" I've added a section for this in the google doc, please, explain this either here or there.

Reply:
I checked the code and the work flow of CSME. Now, when a FW is transferred to ROM at boot up stage or resuming stage, ROM would check the FW with CSME, do some job and reply a msg to HOST to notify it is successful. 

Actually, you could find "error: rx list empty but received " in demsg at boot up stage or resuming stage. It is this case.